walkeeper/src/replication.rs

```diff
@@ -202,10 +220,30 @@ impl ReplicationConn {
 
         let mut end_pos = Lsn(0);
         let mut wal_file: Option<File> = None;
+        let mut retry_count = 0;
 
         loop {
+            if let Ok(lsn) = rx.try_recv() {
```

Comment:
It is slightly dangerous to use channel here: currently pageserver only sends feedback on each message, but it is more efficient to do that only once in a while, like in pg walreceiver. And in that case channel might overflow which will block feedback thread which (eventually) will block pageserver walreceiver. Putting it atomic/under mutex would do.